lib/codegen/model.ejs

```diff
@@ -24,20 +24,29 @@ function printMethod(modelName) {
     return method + params.join(', ') + ') {';
 }
 
+function joinParams(params) {
+  const newParams = params.map(param => `normalize(${param})`);
+  return newParams.join(',');
+}
+
 function printOperation() {
     var serviceName = '_soapModel';
     var params = op.accepts.map(function(a) {
         return a.arg;
     });
     var method = serviceName + '.' + op.operationId
-            + '(' + params.join(', ') + ', function (err, response) {';
+            + '(' + joinParams(params) + ', function (err, response) {';
     return method;
 }
 
 %>
 'use strict';
 var server = require('../../server/server');
 
+function normalize(parameter) {
```

Comment:
This function is made available in generated file, and can assess, at runtime,  if operation parameter object actually has toJSON() method.